lua/framework/physics/field_function/physics_export_field_func.cc

```diff
@@ -29,6 +30,7 @@ ExportFieldFunctionToVTK(lua_State* L)
   //  ff->ExportToVTK(base_name);
   FieldFunctionGridBased::ExportMultipleToVTK(base_name, {ff});
 
+  opensn::mpi_comm.barrier();
```

Comment:
The same business here. Move the barrier inside ` FieldFunctionGridBased::ExportMultipleToVTK`. This way you fix all the usage of that method.